src/pykiso/lib/connectors/trc_handler.py

```diff
@@ -0,0 +1,322 @@
+##########################################################################
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+TRC file handler based on Python CAN
+************************************
+
+:module: trc_handler
+
+:synopsis: Additional layer for python-can trc reader and write to handle can FD
+
+.. currentmodule:: trc_handler
+
+"""
+import logging
+from typing import List, Optional
+from datetime import datetime, timedelta
+
+try:
+    from can.io.trc import TRCFileVersion, TRCReader, TRCWriter
+    from can.util import channel2int, dlc2len, len2dlc
+    from can import Message
+except ImportError as e:
+    raise ImportError(
+        f"{e.name} dependency missing, consider installing pykiso with 'pip install pykiso[can]'"
+    )
+
+
+log = logging.getLogger(__name__)
+
+
+class TypedMessage(Message):
+    """Message with type attribut added and that can handle empty arbitration id.
+    Necessary because python-can does not handle types from CAN FD and informations
+    contained in Message are not enough to deduce it.
+    Some of those types (ex: ST) also have no arbitration id which cannot be handle by
+    python-can Message object.
+    """
+
+    def __init__(self, type: str, msg: Message):
+        """Create a typed message
+
+        :param type: type of CAN messahe (ex:DT)
+        :param msg: python-can message from which to create the typed message
+        """
+        super().__init__()
+        self.type = type
+        self.timestamp = msg.timestamp
+        self.arbitration_id = msg.arbitration_id
+        self.is_extended_id = msg.is_extended_id
+        self.is_remote_frame = msg.is_remote_frame
+        self.is_error_frame = msg.is_error_frame
+        self.channel = msg.channel
+        self.dlc = msg.dlc
+        self.data = msg.data
+        self.is_fd = msg.is_fd
+        self.is_rx = msg.is_rx
+        self.bitrate_switch = msg.bitrate_switch
+        self.error_state_indicator = msg.error_state_indicator
+        self._check = msg._check
+
+    def __repr__(self) -> str:
+        """String representation that can handle messages with no arbitration id
+
+        return: string representation of the message"""
+        if self.type not in ["EC", "ER", "ST"]:
+            args = [
+                f"timestamp={self.timestamp}",
+                f"arbitration_id={self.arbitration_id:#x}",
+                f"is_extended_id={self.is_extended_id}",
+            ]
+        else:
+            args = [
+                f"timestamp={self.timestamp}",
+                f"arbitration_id={self.arbitration_id}",
+                f"is_extended_id={self.is_extended_id}",
+            ]
```

Comment:
Since the arbitration_id is the only difference, maybe this can be shortened, something like:
`args = [
        f"timestamp={self.timestamp}",
        f"arbitration_id={self.arbitration_id if self.type in ["EC", "ER", "ST"] else self.arbitration_id:#x}",
        f"is_extended_id={self.is_extended_id}",
]`